save-orchestrator/src/main/kotlin/org/cqfn/save/orchestrator/service/DockerService.kt

```diff
@@ -168,14 +187,42 @@ class DockerService(private val configProperties: ConfigProperties) {
         )
         saveAgent.delete()
         saveCli.delete()
-        return Pair(imageId, runCmd)
+        return Triple(imageId, agentRunCmd, saveCliExecFlags)
+    }
+
+    private fun collectStandardTestSuitesForDocker(testSuiteDtos: List<TestSuiteDto>?): MutableList<TestSuiteDto> {
+        val testSuitesForDocker: MutableList<TestSuiteDto> = mutableListOf()
+        testSuiteDtos?.forEach {
```

Comment:
But in this method there are two `forEach`:
We go through of all `testSuites`, selected by user and for each of them we've got the another list with standard test suites, we which collect.

So I don't really now how to use `flatMap` in this case, I did some experiments, but without success

Reply:
It's basically what `flatMap` is for: inside a loop you create lists and then should construct a new list out of all the elements. I think it should work like this:
```kotlin
 testSuiteDtos?.flatMap {
    webClientBackend.get()
                .uri("/standardTestSuitesWithName?name=${it.name}")
                .retrieve()
                .bodyToMono<List<TestSuiteDto>>()
                .block()
}
```

Reply:
Thanks, it almost worked, but I was need to additionally use `!!` after `block()`, since I got the
```kotlin\
```

Reply:
Thanks, it almost worked, but I was need to add `!!` after `block()`, because of:

```kotlin
Required:
(TestSuiteDto) → Iterable<TestSuiteDto>
Found:
(TestSuiteDto) → List<TestSuiteDto>?
```
It's little bit strange, since `getStandardTestSuites()` returns not nullable list

Reply:
Yes, that's because `block()` is annotated with `@Nullable` and returns `null` if mono completes empty